src/main/java/de/tum/in/www1/artemis/service/RepositoryService.java

```diff
@@ -28,10 +28,13 @@ public class RepositoryService {
 
     private final GitService gitService;
 
+    private final FileService fileService;
```

Comment:
The `FileService` is large and has quite a few dependencies. This introduces one such dependency, essentially for a single helper function. 

What do you think? It doesn't have to be within this PR, but maybe we can move `FileService.removeIllegalCharacters` to some other class to decouple a bit?